src/hooks/useStepReturnInfo.ts

```diff
@@ -0,0 +1,43 @@
+import { Web3Provider } from '@ethersproject/providers'
+import { Chain, getChainById, Step, Token, TokenAmount } from '@lifinance/sdk'
+import { useWeb3React } from '@web3-react/core'
+import BigNumber from 'bignumber.js'
+import { useEffect, useState } from 'react'
+
+import LiFi from '../LiFi'
+
+interface stepReturnInfo {
+  receivedToken: Token
+  receivedAmount: BigNumber
+  totalBalanceOfReceivedToken: TokenAmount | null
+  receivedTokenMatchesPlannedToken: boolean
+  toChain: Chain
+}
+
+export const useStepReturnInfo = (step: Step) => {
+  const web3 = useWeb3React<Web3Provider>()
+  const [stepReturnInfo, setStepReturnInfo] = useState<stepReturnInfo>()
+
+  useEffect(() => {
+    const getInfo = async () => {
```

Comment:
I think this should be declared outside of the `useEffect` by using a `useCallback` hook. This allows for function redefinition only when one of the fn's dependencies change